actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -600,17 +600,20 @@ def match(path, options=nil)
         def mount(app, options = nil)
           if options
             path = options.delete(:at)
-          else
-            unless Hash === app
-              raise ArgumentError, "must be called with mount point"
-            end
```

Comment:
This short-cut did not validate `app` is callable when `mapper.mount 10, as: "exciting"`.